### PR TITLE
Read notifications

### PR DESCRIPTION
**Pull request by 30bySixty**
Implement feature to mark one or all notifications as read.

### CODE REVIEW

**Comment by 30bySixty** [comment]
[AAU, I can mark individual notifications as read.](https://trello.com/c/RhSTITn7/116-aau-i-can-mark-individual-notifications-as-read)

**Comment by 30bySixty** [comment]
[AAU, I can mark all notifications as read.](https://trello.com/c/rrf9qFZc/117-aau-i-can-mark-all-notifications-as-read)

File: app/controllers/dashboard_controller.rb [hunk 1]
```diff
@@ -44,7 +44,8 @@ def followings
   
   def notifications
     respond_to do |format|
-      format.js {render :partial => 'dashboard/notifications'} 
+      format.js {render :partial => 'dashboard/notifications'}
```
Comment (wenga86) [inline app/controllers/dashboard_controller.rb]: A important thing to note is that if the mime type is not provided in the request, rails will serve the first format that in the block so if your want your default format to be `.js` 👍

File: app/controllers/notifications_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class NotificationsController < ApplicationController
+  def mark_as_read
+    @user = current_user
+    @notification = Notification.find(params[:id])
+    @notification.update read: true
```
Comment (wenga86) [inline app/controllers/notifications_controller.rb]: If you want you can merge the `read` attribute into your params so you only have to make one update to the database.

File: app/controllers/notifications_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        format.html {redirect_to notifications_dashboard_path(@user) }
+        format.js {render :partial => 'dashboard/notifications/markasread', :data => @notification.to_json}
+      end
+    end
+  end
+  
+  def mark_all_as_read
+    @user = current_user
+    @unread_notifications = Notification.where(user_id: @user.id, read: false)
+    @unread_notifications.each do |notification|
```
Comment (wenga86) [inline app/controllers/notifications_controller.rb]: I would recommend update all here if you don't require callbacks from the app. It's a lot faster and updates all records in a single query.
```
@unread_notifications = Notification.where(user_id: @user.id, read: false).update_all(read: true)
```
http://apidock.com/rails/v4.0.2/ActiveRecord/Relation/update_all

**Comment by adayag** [comment]
looks good. you might want more unit testing on your controllers

**Pull request merged**